File: app/src/main/java/com/fengniao/news/net/HttpUtils.java
package com.fengniao.news.net;

import android.text.TextUtils;

import okhttp3.FormBody;

import static com.fengniao.news.net.Api.URL_NEWS_BEFORE_ZHIHU;
import static com.fengniao.news.net.Api.URL_NEWS_TOU_TIAO;

public class HttpUtils {
    private static HttpUtils mHttpUtils;
    private FormBody.Builder mBuilder;

    private HttpUtils() {
    }

    public static HttpUtils getInstance() {
        if (mHttpUtils == null) {
            mHttpUtils = new HttpUtils();
        }
        return mHttpUtils;
    }

    //添加公共参数
    public FormBody.Builder getBaseFormBodyBuilder() {
        if (mBuilder == null)
            mBuilder = new FormBody.Builder();
        return mBuilder;
    }

    public void setToken(String token) {
        if (!TextUtils.isEmpty(token))
            getBaseFormBodyBuilder().add("token", token);
    }


    //获取头条新闻
    public void getTouTiaoNews(FNCallback callback) {
        OkHttpManager.getInstance().getAsyn(URL_NEWS_TOU_TIAO, new FNCallbackHolder(URL_NEWS_TOU_TIAO, callback));
    }

    //获取知乎文章
    public void getZhiHuArticle(String date, FNCallback callback) {
        OkHttpManager.getInstance().getAsyn(URL_NEWS_BEFORE_ZHIHU + date, new FNCallbackHolder(URL_NEWS_BEFORE_ZHIHU, callback));
    }

}

File: app/src/main/java/com/fengniao/news/net/HttpUtils.kt
package com.fengniao.news.net

import android.text.TextUtils
import com.fengniao.news.net.ApiConstants.URL_NEWS_BEFORE_ZHIHU
import com.fengniao.news.net.ApiConstants.URL_NEWS_TOU_TIAO
import okhttp3.FormBody

class HttpUtils private constructor() {
    private var token: String? = null

    //添加公共参数
    val baseFormBodyBuilder: FormBody.Builder
        get() {
            val builder = FormBody.Builder()
            if (!TextUtils.isEmpty(token))
                builder.add("token", token!!)
            return builder
        }

    fun setToken(token: String) {
        this.token = token
    }


    //获取头条新闻
    fun getTouTiaoNews(callback: FNCallback) {
        OkHttpManager.instance.getAsyn(URL_NEWS_TOU_TIAO, FNCallbackHolder(URL_NEWS_TOU_TIAO, callback))
    }

    //获取知乎文章
    fun getZhiHuArticle(date: String, callback: FNCallback) {
        OkHttpManager.instance.getAsyn(URL_NEWS_BEFORE_ZHIHU + date, FNCallbackHolder(URL_NEWS_BEFORE_ZHIHU, callback))
    }

    companion object {
        private var mHttpUtils: HttpUtils? = null

        val instance: HttpUtils
            get() {
                if (mHttpUtils == null) {
                    mHttpUtils = HttpUtils()
                }
                return mHttpUtils!!
            }
    }

}
